controllers/main.go

```diff
@@ -28,6 +30,40 @@ import (
 	"gopkg.in/gorp.v1"
 )
 
+// ipRange is a structure that holds the start and end of a range of ip addresses
+type ipRange struct {
```

Comment:
Wouldn't it be more appropriate/standard to use net.IPNet instead?  Then checking whether an IP is in a network is just `ipNet.Contains(ip)`.